build/BuildSteps.cs

```diff
@@ -44,6 +39,56 @@ public static void RestorePackages()
             Shell.Run("dotnet", $"restore {Settings.ProjectFile} {feeds}");
         }
 
+        public static void UpdatePackageVersionForIntegrationTests()
+        {
+            if (string.IsNullOrEmpty(Settings.IntegrationBuildNumber))
+            {
+                ColoredConsole.Error.WriteLine($"Environment variable 'integrationBuildNumber' cannot be null or empty for an integration build.");
+            }
+
+            var feed = "https://azfunc.pkgs.visualstudio.com/e6a70c92-4128-439f-8012-382fe78d6396/_packaging/AzureFunctionsPreRelease/nuget/v3/index.json";
+            var packagesToUpdate = GetV3PackageList();
+            string currentDirectory = null;
+
+            Dictionary<string, string> buildPackages = new Dictionary<string, string>();
+
+            _manifest = new IntegrationTestBuildManifest();
+
+            try
+            {
+                currentDirectory = Directory.GetCurrentDirectory();
+                var projectFolder = Path.GetFullPath(Settings.SrcProjectPath);
+                Directory.SetCurrentDirectory(projectFolder);
+
+                foreach (var package in packagesToUpdate)
+                {
+                    string packageInfo = Shell.GetOutput("NuGet", $"list {package} -Source {feed} -prerelease").Split(Environment.NewLine)[0];
+
+                    if (string.IsNullOrEmpty(packageInfo))
+                    {
+                        ColoredConsole.Error.WriteLine($"Failed to get {package} package information from {feed}.");
```

Comment:
If list fails, restore would fail too. Should skip adding the package incase list fails or and stop the build pipeline if this happens.

Reply:
Yes, I agree. I have updated the code, so that if cannot get the package info, we throw and stop the build execution.